src/lib.rs

```diff
@@ -556,7 +555,7 @@ mod tests {
         assert_eq!(queue.iter().collect::<Vec<_>>(), vec![&1, &5, &2, &50]);
         println!("Same elements + 50: {:?}", queue.iter().collect::<Vec<_>>());
 
-        sleep_secs(2);
+        sleep_secs(1);
```

Comment:
Over here, maybe switching to a fake clock for the tests can make the tests deterministic but that would increase complexity a bit.

Reply:
Actually this test was wrong, waiting 2 sec should make the element to expire and your improvements and the change here put in evidence that.